src/proxy-animation.js

```diff
@@ -228,23 +544,32 @@ export class ProxyAnimation {
 
   get ready() {
     const details = proxyAnimations.get(this);
-    if (details.readyPromise)
-      return details.readyPromise;
+    if (details.timeline) {
+      if (details.readyPromise)
+        return details.readyPromise;
+
+      // TODO: If not waiting on a pending task, we still need to return a ready
+      // promise; however, the promise can be immediately resolved. Cannot use
+      // the underlying animation as it is intentionally locked in a
+      // pause-pending state.
+    }
 
-    proxyAnimations.get(this).animation.ready;
+    return details.animation.ready;
   }
 
 };
 
 export function animate(keyframes, options) {
   let timeline = options.timeline;
   if (!timeline || !(timeline instanceof ScrollTimeline)) {
-    return nativeElementAnimate.apply(this, [keyframes, options]);
+    let animation = nativeElementAnimate.apply(this, [keyframes, options]);
+    // Even through this animation runs as a native animation, we still wrap
+    // it in a proxy animation to allow changing of the animation's timeline.
```

Comment:
Maybe this shouldn't be a special case anymore, i.e. this method can be

if (timeline instanceof ScrollTimeline)
  delete options.timeline;

// common animation construction.

if (timeline instanceof ScrollTimeline) {
  animation.pause();
  proxyAnimation.play();
}